### PR TITLE
Ceph fix object bucket claim cleanup

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**
Creating this as a draft PR because there are multiple issues to fix for #6550 

Currently bulding on top of #6568 

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by BlaineEXE** [comment]
I'm not sure the OBC code that allows generating names has worked for 1+ years if it has ever worked.

https://github.com/kube-object-storage/lib-bucket-provisioner/pull/43/files#r526501050

File: pkg/operator/ceph/object/bucket/provisioner.go [hunk 1]
```diff
@@ -385,7 +385,8 @@ func (p *Provisioner) composeObjectBucket() *bktv1alpha1.ObjectBucket {
 
 	return &bktv1alpha1.ObjectBucket{
 		Spec: bktv1alpha1.ObjectBucketSpec{
-			Connection: conn,
+			StorageClassName: storageClassName,
```
Comment (thotz) [inline pkg/operator/ceph/object/bucket/provisioner.go:388]: The lib bucket provisioner sets storageclass name in https://github.com/kube-object-storage/lib-bucket-provisioner/blob/master/pkg/provisioner/controller.go#L383 after `Provisioner` or `Grant`. So this change may be reductant

Comment (BlaineEXE) [reply pkg/operator/ceph/object/bucket/provisioner.go:388]: That is only set **after** the Secret and ConfigMap are created, so if those or anything before it fails, the cleanup function will have empty information and fail to clean up the resource properly.

Comment (travisn) [reply pkg/operator/ceph/object/bucket/provisioner.go:388]: From your description sounds like a good thing to update the lib-bucket-provisioner, but I haven't looked in more detail.

**Comment by BlaineEXE** [comment]
Per chat with Travis, closing this in favor of fixing in lib-bucket-provisioner.

**Pull request closed without merging**